### PR TITLE
Fix various small compile errors

### PR DESCRIPTION
**Pull request by p2004a**
I don't get why gcc is fine with those mistakes...

### CODE REVIEW

**Comment by lhog** [comment]
What tool did you use to catch these?

File: rts/System/ScopedResource.h [hunk 1]
```diff
@@ -37,12 +37,6 @@ namespace spring {
 			if (!released) d(r);
 		}
 
-		const R&& Release() const {
```
Comment (lhog) [inline rts/System/ScopedResource.h:40]: This one might still make sense w/o the first `const`?

Comment (p2004a) [reply rts/System/ScopedResource.h:40]: It tries to modify `released`, so unless `released` is mutable, you can't do that.

**Comment by p2004a** [comment]
It failed to compile under clang with those errors present. I use clangd for autocompletion, so I care that it compiles under clang.

**Comment by lhog** [comment]
Very well, thanks!

**Pull request merged**